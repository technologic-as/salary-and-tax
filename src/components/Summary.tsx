import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import React from 'react';
import { defineMessages, useIntl } from 'react-intl';
import { connect } from 'react-redux';
import { Section, TaxRow } from './Ui';
import { State } from '../reducers';

const messages = defineMessages({
  header: { id: 'summary.header', defaultMessage: 'Summary' },
  turnover: { id: 'summary.turnover', defaultMessage: 'Turnover' },
  totalCuts: { id: 'summary.total.cuts', defaultMessage: 'Income' },
  totalTax: { id: 'summary.total.tax', defaultMessage: 'Total tax' },
  totalIncome: { id: 'summary.total.income', defaultMessage: 'Total income' },
  income: { id: 'summary.income', defaultMessage: 'Income' },
  dividends: { id: 'summary.dividends', defaultMessage: 'Dividends' },
});

type Props = {
  income: number;
  totalTax: number;
  afterDividendsTax: number;
  totalCuts: number;
  afterTotal: number;
  afterTax: number;
  turnover: number;
};

export const SummaryComponent = ({
  turnover,
  income,
  totalTax,
  afterTax,
  totalCuts,
  afterDividendsTax,
  afterTotal,
}: Props): JSX.Element => {
  const { formatNumber, formatMessage } = useIntl();
  const formatCurrency = (amount: number) => formatNumber(amount, { style: 'currency', currency: 'NOK' });
  return (
    <Section header={formatMessage(messages.header)} expanded>
      <Table>
        <TableBody>
          <TaxRow description={formatMessage(messages.turnover)} sum={formatCurrency(turnover)} />
          <TaxRow
            description={formatMessage(messages.totalCuts)}
            amount={formatCurrency(totalCuts)}
            sum={formatCurrency(income)}
            minus
          />
          <TaxRow
            description={formatMessage(messages.totalTax)}
            amount={formatCurrency(totalTax)}
            sum={formatCurrency(afterTax)}
            minus
          />
          <TaxRow description={formatMessage(messages.income)} amount={formatCurrency(afterTax)} />
          <TaxRow description={formatMessage(messages.dividends)} amount={formatCurrency(afterDividendsTax)} />
          <TaxRow description={formatMessage(messages.totalIncome)} sum={formatCurrency(afterTotal)} />
        </TableBody>
      </Table>
    </Section>
  );
};

const mapStateToProps = ({
  calculations: {
    tax: { totalTax, afterTax },
    salary: { turnover, income, totalCuts },
    dividends: { afterDividendsTax },
    afterTotal,
  },
}: State) => ({
  turnover,
  income,
  totalTax,
  afterTax,
  totalCuts,
  afterDividendsTax,
  afterTotal,
});

const mapDispatchToProps = () => ({});

export const Summary = connect(mapStateToProps, mapDispatchToProps)(SummaryComponent);

export default Summary;
