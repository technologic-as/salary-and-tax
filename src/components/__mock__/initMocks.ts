const mockFormatMessage = ({ defaultMessage }: { defaultMessage: string }) => defaultMessage;
const mockFormatNumber = (amount: number) =>
  new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD' }).format(amount);

const mockIntl = () => ({
  formatMessage: mockFormatMessage,
  formatDate: jest.fn,
  formatTime: jest.fn,
  formatRelative: jest.fn,
  formatNumber: mockFormatNumber,
  formatPlural: jest.fn,
  formatHTMLMessage: jest.fn,
  now: jest.fn,
});

jest.mock('react-intl', () => ({
  useIntl: mockIntl,
  defineMessages: (msg: unknown) => msg,
}));

jest.mock('react-intl-redux', () => ({
  updateIntl: (a: unknown) => a,
}));

jest.mock('@material-ui/core/Table', () => 'Table');
jest.mock('@material-ui/core/TableBody', () => 'TableBody');
jest.mock('@material-ui/core/FormGroup', () => 'FormGroup');
jest.mock('@material-ui/core/FormControlLabel', () => 'FormControlLabel');
jest.mock('@material-ui/core/Radio', () => 'Radio');

export {};
