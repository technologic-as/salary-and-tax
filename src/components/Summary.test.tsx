import mockComponent from 'identity-obj-proxy';
import React from 'react';
import renderer from 'react-test-renderer';
import './__mock__/initMocks';
import { SummaryComponent } from './Summary';

jest.mock('./Ui', () => mockComponent);
jest.mock('@material-ui/core/Table', () => 'Table');
jest.mock('@material-ui/core/TableBody', () => 'TableBody');
jest.mock('@material-ui/core/TableHead', () => 'TableHead');

describe('Summary', () => {
  it('should render', () => {
    const tree = renderer
      .create(
        <SummaryComponent
          turnover={111111}
          income={222222}
          totalTax={444444}
          afterTax={555555}
          totalCuts={666666}
          afterDividendsTax={777777}
          afterTotal={888888}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
