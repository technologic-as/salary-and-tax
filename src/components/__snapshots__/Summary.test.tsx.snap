// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary should render 1`] = `
<Section
  expanded={true}
  header="Summary"
>
  <Table>
    <TableBody>
      <TaxRow
        description="Turnover"
        sum="$111,111.00"
      />
      <TaxRow
        amount="$666,666.00"
        description="Income"
        minus={true}
        sum="$222,222.00"
      />
      <TaxRow
        amount="$444,444.00"
        description="Total tax"
        minus={true}
        sum="$555,555.00"
      />
      <TaxRow
        amount="$555,555.00"
        description="Income"
      />
      <TaxRow
        amount="$777,777.00"
        description="Dividends"
      />
      <TaxRow
        description="Total income"
        sum="$888,888.00"
      />
    </TableBody>
  </Table>
</Section>
`;
